opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -538,6 +538,21 @@ def use_span(
                 self.source._current_span_slot.set(  # pylint:disable=protected-access
                     span_snapshot
                 )
+
+        except Exception as error:  # pylint: disable=broad-except
+            if (
```

Comment:
Is it possible to used the logic already existing in `Span.__exit__()`?

Reply:
I tried to, but `Span.__exit__` includes a call to `super` that ends up calling `end` which then raises a warning because of this method being called on an already ended `Span`. I find the code here to be sufficiently different from `Span.__exit__` (because here it is not required to check for an exception being present and we don't call `super`) to make it ok to have similar code here.